File: 1ºAno/Lista3/C/Casa/files/ex10.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

int main() {
    for (;;) {
        char op[2];
        double vals[2] = {0, 0};
        for (;;) {
            printf ("Escolha uma de nossas operações suportadas:\n+  -  *  /  ^  S\nOP = ");
            scanf ("%s", op);
            getchar();
            if (op[0] != '+' && op[0] != '-' && op[0] != '*' && op[0] != '/' && op[0] != '^' && op[0] != 'S' && op[0] != 's') {
                printf ("*Operação inválida... Tente novamente*");
            }
            else {
                break;
            }
        }
        if (op[0] == 'S' || op[0] == 's') {
            break;
        }
        printf ("Na mesma linha, entre com os dois números para a realização da operação:\n");
        scanf ("%lf %lf", &vals[0], &vals[1]);
        if (op[0] == '+') {
            printf ("%.2lf + %.2lf = %.2lf\n\n", vals[0], vals[1], vals[0] + vals[1]);
        }
        else if (op[0] == '-') {
            printf ("%.2lf - %.2lf = %.2lf\n\n", vals[0], vals[1], vals[0] - vals[1]);
        }
        else if (op[0] == '*') {
            printf ("%.2lf * %.2lf = %.2lf\n\n", vals[0], vals[1], vals[0] * vals[1]);            
        }
        else if (op[0] == '/') {
            printf ("%.2lf / %.2lf = %.2lf\n\n", vals[0], vals[1], vals[0] / vals[1]);            
        }
        else if (op[0] == '^') {
            printf ("%.2lf ^ %.2lf = %.2lf\n\n", vals[0], vals[1], pow(vals[0], vals[1])); 
        }
    }
    printf ("\nAdeus!");
    return 0;
}

File: 1ºAno/Lista3/Python/Casa/files/ex10.py
while True:
    while True:
        op = input("Escolha uma entre nossas operações suportadas:\n+  -  *  /  ^  S\nOperação = ")
        if op != "+" and op != "-" and op != "*" and op != "/" and op != "**" and op != "^" and op != "S" and op != "s":
            print ("Operação inválida... Tente novamente")
        else:
            break
    if op == "s" or op == "S":
        break
    while True:
        try:
            nums = input("Na mesma linha, entre com os dois número para a realização da operação escolhida:\n").split()
            nums[0], nums[1] = int(nums[0]), int(nums[1])
        except:
            print ("Número(s) inválido(s)... Tente novamente")
        else:
            break
    if op == "+":
        print ("{} + {} = {:.2f}\n" .format(nums[0], nums[1], nums[0] + nums[1]))
    elif op == "-":
        print ("{} - {} = {:.2f}\n" .format(nums[0], nums[1], nums[0] - nums[1]))
    elif op == "*":
        print ("{} * {} = {:.2f}\n" .format(nums[0], nums[1], nums[0] * nums[1]))
    elif op == "/":
        print ("{} / {} = {:.2f}\n" .format(nums[0], nums[1], nums[0] / nums[1]))
    elif op == "**" or op == "^":
        print ("{} ^ {} = {:.2f}\n" .format(nums[0], nums[1], nums[0] ** nums[1]))
print ("\nAdeus!")